push/src/main/java/org/aerogear/mobile/push/PushService.java

```diff
@@ -0,0 +1,250 @@
+    private static final List<MessageHandler> BACKGROUND_THREAD_HANDLERS = new ArrayList<>();
+
+    private final String deviceType = "ANDROID";
+    private final String operatingSystem = "android";
+    private final String osVersion = android.os.Build.VERSION.RELEASE;
+
+    private MobileCore core;
+    private String url;
+    private UnifiedPushCredentials unifiedPushCredentials;
+
+    @Override
+    public String type() {
+        return "push";
+    }
+
+    @Override
+    public void configure(MobileCore core, ServiceConfiguration serviceConfiguration) {
+        this.core = core;
+        this.url = serviceConfiguration.getUrl();
+        loadConfigJson();
+    }
+
+    @Override
+    public boolean requiresConfiguration() {
+        return true;
+    }
+
+    @Override
+    public void destroy() {}
+
+    private void loadConfigJson() {
+        InputStream fileStream = null;
+
+        try {
+            fileStream = core.getContext().getResources().getAssets().open(PUSH_CONFIG_FILE_NAME);
+            int size = fileStream.available();
+            byte[] buffer = new byte[size];
+            fileStream.read(buffer);
+            fileStream.close();
+            String json = new String(buffer);
+
+            JSONObject pushConfig = new JSONObject(json);
+            JSONObject pushAndroidConfig = pushConfig.getJSONObject(JSON_OBJECT);
+            UnifiedPushCredentials unifiedPushCredentials = new UnifiedPushCredentials();
+            unifiedPushCredentials.setSender(pushAndroidConfig.getString(JSON_SENDER_ID));
+            unifiedPushCredentials.setVariant(pushAndroidConfig.getString(JSON_VARIANT_ID));
+            unifiedPushCredentials.setSecret(pushAndroidConfig.getString(JSON_VARIANT_SECRET));
+
+            this.unifiedPushCredentials = unifiedPushCredentials;
+        } catch (JSONException e) {
+            MobileCore.getLogger().error(e.getMessage(), e);
+            throw new RuntimeException("An error occurred while parsing the "
+                            + PUSH_CONFIG_FILE_NAME + ". Please check the file format");
+        } catch (IOException e) {
+            MobileCore.getLogger().error(e.getMessage(), e);
+            throw new RuntimeException("An error occurred while parsing the "
+                            + PUSH_CONFIG_FILE_NAME + ". Please check if the file exists");
+        } finally {
+            if (fileStream != null) {
+                try {
+                    fileStream.close();
+                } catch (IOException e) {
+                    // Ignore IOException
+                }
+            }
+        }
+    }
+
+    public void registerDevice(final Callback callback) {
+        registerDevice(new UnifiedPushConfig(), callback);
+    }
+
+    public void registerDevice(final UnifiedPushConfig unifiedPushConfig, final Callback callback) {
+
+        try {
+
+            JSONObject data = new JSONObject();
```

Comment:
I've seen JSONObjects constructed this way a lot.  We should probably look at making a utility class to make them easier to do

Idea.java
```java
JSONUtils.object(
  propery("deviceToken", FirebaseInstanceId.getInstance().getToken()),
  propery("deviceType", deviceType)
  //etc
).build();
```

Reply:
I like this too